crates/database/src/solver_competition.rs

```diff
@@ -1,29 +1,30 @@
-use crate::TransactionHash;
+use crate::{auction::AuctionId, TransactionHash};
 use sqlx::{types::JsonValue, PgConnection};
 
-pub type SolverCompetitionId = i64;
-
 pub async fn save(
     ex: &mut PgConnection,
+    id: AuctionId,
     data: &JsonValue,
     tx_hash: Option<&TransactionHash>,
-) -> Result<SolverCompetitionId, sqlx::Error> {
+) -> Result<(), sqlx::Error> {
     const QUERY: &str = r#"
-INSERT INTO solver_competitions (json, tx_hash)
-VALUES ($1, $2)
-RETURNING id
-    "#;
-    let (id,) = sqlx::query_as(QUERY)
+INSERT INTO solver_competitions (id, json, tx_hash)
+VALUES ($1, $2, $3)
+ON CONFLICT (id) DO UPDATE
```

Comment:
Why is this needed now (and wasn't needed before)? Wouldn't this signify that we accidentally reused an auction ID somehow?

Reply:
🤦 - explanation ☝️